<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>665</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>781</width>
      <height>601</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Maze</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_open">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>20</y>
        <width>191</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string>Load maze from file</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>130</y>
        <width>191</width>
        <height>181</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="pushButton_eller">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>171</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>Generate 
a perfect maze</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>58</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string>colums</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_rows">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>10</y>
         <width>52</width>
         <height>32</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>20</y>
         <width>58</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string>rows:</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_cols">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>50</y>
         <width>52</width>
         <height>32</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
     </widget>
     <widget class="Painter" name="widget" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>550</width>
        <height>550</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background: gray</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame_2">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>370</y>
        <width>191</width>
        <height>181</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="pushButton_path">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>171</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Show the solution</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_from_row">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>20</y>
         <width>52</width>
         <height>32</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_from_col">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>20</y>
         <width>52</width>
         <height>32</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>58</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string>from:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>58</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string>to:</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_to_row">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>60</y>
         <width>52</width>
         <height>32</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_to_col">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>60</y>
         <width>52</width>
         <height>32</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>74</x>
         <y>0</y>
         <width>31</width>
         <height>18</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>row:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>0</y>
         <width>58</width>
         <height>18</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>colum:</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Painter</class>
   <extends>QWidget</extends>
   <header location="global">painter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
